lib/api/client-commands/_base-command.ts

```diff
@@ -1,5 +1,22 @@
-module.exports = class ClientCommand {
-  static get isTraceable() {
+interface ResultObject {
+  status: number;
+  code: any;
+  name: any;
+  value: {
+    message: any;
+  };
+  error: any;
+}
+
+interface MakePromiseOptions {
+  performAction: (callback : (result : ResultObject) => void) => void;
+  userSuppliedCallback?: (param : any) => any;
+  fullResultObject?: boolean;
+  fullPromiseResolve?: boolean;
+}
```

Comment:
We should create an interface for the parameters of a function only if we want to reuse it, otherwise just put the type beside each property in parameter:
```ts
static makePromise({
      performAction: (callback: (result: ResultObject) => void) => void,
      userSuppliedCallback: ((param : any) => any) = function() {}, 
      fullResultObject: boolean = true, 
      fullPromiseResolve: boolean = true
    }) : Promise<unknown> {}
```